src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.liquid

```diff
@@ -267,9 +267,9 @@
 
                     ProcessResponse(client_, response_);
 
-                    var status_ = ((int)response_.StatusCode).ToString();
+                    var status_ = (int)response_.StatusCode;
 {%     for response in operation.Responses -%}
-                    if (status_ == "{{ response.StatusCode }}"{% if response.CheckChunkedStatusCode %} || status_ == "206"{% endif %}) 
+                    if (status_ == {{ response.StatusCode }}{% if response.CheckChunkedStatusCode %} || status_ == 206{% endif %})
```

Comment:
This here is exactly the problem why it's a string => The OpenAPI status code might not be a number but "2XX" or "default" which now might lead to compile errors: 

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#responses-object

So now we probably need to fix that.